src/wp-includes/post.php

```diff
@@ -2658,7 +2658,7 @@ function unstick_post( $post_id ) {
 		return;
 	}
 
-	$stickies = array_map( 'intval', $stickies );
+	$stickies = array_values( array_unique( array_map( 'intval', $stickies ) ) );
```

Comment:
As above re extra DB calls and actions firing incorrectly.

Reply:
`array_unique` preserves the keys so if you have `[0 => 1, 1 => 1, 2 => 2 ]` the result will be `[0 => 1, 2 => 2 ]` and it won't be possible to remove the second element with actual code:

```php
$offset = array_search( $post_id, $stickies, true );
if ( false === $offset ) {
	return;
}

array_splice( $stickies, $offset, 1 );
```
In the `unstick_post`, I can't see how it'll do an extra DB call.

Reply:
> In the `unstick_post`, I can't see how it'll do an extra DB call.

That's because you're absolutely right, it can't and I was mistaken. 😉 
The `in_array()` check on L2663 protects against unnecessary DB writes.